railties/lib/rails/generators/rails/resource_route/resource_route_generator.rb

```diff
@@ -21,7 +25,9 @@ def add_resource_route
         end
 
         # inserts the primary resource
-        write("resources :#{file_name.pluralize}", route_length + 1)
+        resource_string = "resources :#{file_name.pluralize}"
+        resource_string << ", except: [:new, :edit]" if options.api?
```

Comment:
I tried investigating this, and did not find an easy way to achieve it. api_only is made available via config, there is no such easy identifiable(and sure) way for engines, when they are being mounted.